src/service-implementations/lf-repo-tree-node-service.ts

```diff
@@ -0,0 +1,357 @@
+      if (!path) {
+        throw new Error('fullPath is undefined');
+      }
+    }
+
+    const folderNode: LfRepoTreeNode = {
+      name: entryName,
+      path: path,
+      id: entry.id!.toString(),
+      parentId: parent?.id,
+      entryType: entry.entryType,
+      icon: [],
+      isContainer: true,
+      isLeaf: false,
+      attributes: new Map<string, any>()
+    };
+    return folderNode;
+  }
+
+  private async getFolderChildrenFirstPageAsync(folder: LfRepoTreeNode): Promise<ODataValueContextOfIListOfEntry> {
+    let entryId: number;
+    if (folder.targetId) {
+      entryId = folder.targetId;
+    }
+    else {
+      entryId = parseInt(folder.id, 10);
+    }
+    const repoId: string = await this.repoClient.getCurrentRepoId();
+    const requestParameters = await getFolderChildrenDefaultParametersAsync(repoId, entryId);
+    const listChildrenEntriesResponse: ODataValueContextOfIListOfEntry = await this.repoClient.entriesClient.getEntryListing(
+      requestParameters
+    );
+    return listChildrenEntriesResponse;
+  }
+
+  private parseFolderChildrenResponse(parent: LfRepoTreeNode, repoName: string, listChildrenEntriesResponse: ODataValueContextOfIListOfEntry): LfRepoTreeNode[] {
+    const dataMap: LfRepoTreeNode[] = [];
+    const childrenEntries: Entry[] | undefined = listChildrenEntriesResponse.value;
+    if (childrenEntries) {
+      for (const childEntry of childrenEntries) {
+        if (this.isViewable(childEntry)) {
+          const childNode: LfRepoTreeNode = this.createNode(childEntry, repoName, parent);
+          dataMap.push(childNode);
+        }
+      }
+    }
+    return dataMap;
+  }
+  private async getFolderEntryAsync(path: string): Promise<Entry | undefined> {
+    const folderNames: string[] = PathUtils.getListOfFolderNames(path);
+    const rootId = 1;
+    let parentId: number = rootId;
+    let foundParentFolder: Entry;
+    let needToCreateRestOfFolders: boolean = false; // Stop checking for existing subfolder if we know it won't exist (fewer API calls)
+
+    for (const folderName of folderNames) {
+      if (!needToCreateRestOfFolders) {
+        const foundSubfolder = await this.findSubfolderByNameAsync(parentId, folderName);
+        if (foundSubfolder) {
+          if (foundSubfolder.id) {
+            let entryId: number;
+            if ((foundSubfolder as Shortcut).targetType === EntryType.Shortcut) {
+              // TODO remove this once the API team allows targetID to be on children select
+              const shortcut = foundSubfolder as Shortcut;
+              entryId = shortcut.targetId;
+            } else {
+              entryId = foundSubfolder.id;
+            }
+            parentId = entryId;
+            foundParentFolder = foundSubfolder;
+            continue;
+          } else {
+            throw new Error('Failed to find folder');
+          }
+        }
+      }
+      needToCreateRestOfFolders = true;
+    }
+    return foundParentFolder;
```

Comment:
update foundParentFolder to have the logical path (path that is passed in)

Reply:
actually leave until createNode, node should have overwritten path

Reply:
@paolo-arge-lf Ke and I went back and forth on this. do you think the logical path should be overwritten here on the entry or overwritten when the entry is converted to an LfRepoTreeNode?

The issue was the createNode function already overwrote the path if a parent existed, so we would have to change the interface of that function